client/state/test/index.js

```diff
@@ -0,0 +1,46 @@
+require( 'lib/react-test-env-setup' )();
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import mockery from 'mockery';
+
+describe( 'state', () => {
+	var createReduxStore;
+	before( () => {
+		mockery.registerMock( 'lib/themes/middlewares.js', {
+			// stub redux middleware
+			// see: http://rackt.org/redux/docs/advanced/Middleware.html
+			analyticsMiddleware: store => next => action => { return next( action ); }
```

Comment:
why `return` kind-of breaks the beautiful chain!

```js
analyticsMiddleware: store => next => action => next( action );
```

of course, isn't the last part of the chain here unnecessary anyway? I think that the call after `next =>` just returns `next`

```js
analyticsMiddleware: store => next => next;
```

Here, `next` gets returned and if you call the returned result, as it is written, it returns a function that accepts a single argument, which it calls `action`, then calls `next( action )`. In my shortened version, it actually returns the `next` function itself and if you call it again and happen to pass `action`, it will return the result of passing `action` into `next`, which is equivalent.